src/pudl/analysis/allocate_net_gen.py

```diff
@@ -235,25 +195,101 @@ def allocate_gen_fuel_by_gen_pm_fuel(gf, gen, gens, drop_interim_cols=True):
                 "fuel_consumed_mmbtu",
             ]
         ]
+    _test_gen_fuel_allocation(gen, gen_pm_fuel_frac)
     return gen_pm_fuel_frac
 
 
+def aggregate_generation_fuel_by_generator(
+    pudl_out, gen_pm_fuel: pd.DataFrame
+) -> pd.DataFrame:
+    """
+    Aggregate gen fuel data columns to generators.
+
+    The generation_fuel_eia923 table includes net generation and fuel
+    consumption data at the plant/fuel type/prime mover level. The most
+    granular level of plants that PUDL typically uses is at the plant/generator
+    level. This function takes the plant/fuel type/prime mover level allocation,
+    aggregates it to the generator level and then denormalizes it to make it
```

Comment:
What do you mean by "denormalizes" here?

Reply:
it means merge in all of the generator and unit and utility information. it is the same process that is used on the non-allocated/original generation table in the outputs